### PR TITLE
java: Add -gen java:default_unsupported

### PR DESCRIPTION
**Pull request by brettkail-wk**
### Story:
For changes to .frugal APIs, some teams rebuild all dependent code (services, clients) to ensure there are no breaking changes.  For service method additions, this rebuild always fails because the handler class does not implement the new method, so the failed rebuild requires manual inspection and approval.

Java supports interface method default implementations, so we can avoid this problem by generating a stub in the interface that throws an unsupported exception to the client.

Commits are intended to be reviewed independently.  The first commit adds a large baseline file equivalent, and the subsequent commit amends it with the option so that the delta can be reviewed.

### Acceptance Criteria:
- [x] Code has been tested and results documented
- [x] Unit tests have been updated
- [ ] Updates to documentation if necessary
- [ ] Verify and document changes to any other Messaging components
- [x] Pull request made against the 'develop' branch, not master

### Design Notes:
There are two minor drawbacks:
1. Increased `.class` file size.  In a repository with ~100 service methods and ~60 structs that generates a ~2.4M JAR, the JAR size increased by ~2.5K (~0.1% increase).  This seems negligibly small.
2. Potential for developer surprise/confusion in not getting compiler errors for unimplemented methods when implementing a new service interface from scratch.  Implementing a handler from scratch is relatively rare, and the surprise seems non-impactful.

To avoid either drawback, this PR implements the default implementations with a new option.

### My Test Results:
I tested manually in the above-mentioned repo: added a new service method, generated, consumed in a service, observed build failure, regenerated with the option, reconsumed in the service, and observed a passing build.

#### Reviewers:
@Workiva/service-platform
@shaunbrockhoff-wk @mitchellsinclair-wk

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
The items listed below may not capture all security relevant changes. Before providing a security review, be sure to review the entire PR for security impact.
<details>
<summary>(2) Security relevant changes were detected</summary>
<li>Watched keyword <code>ObjectOutputStream</code> in <code>compiler/testdata/expected/java/variety_defaultunsupported/FFoo.java</code> line(s) ['1115', '1331', '1765', '2243', '2658', '3071', '3475', '3934', '4265', '4677', '5033', '5273', '5565', '5795', '6086', '6393', '6699', '7002', '7305', '7608', '7911', '8214', '8517'] added</li>
<li>Watched keyword <code>ObjectInputStream</code> in <code>compiler/testdata/expected/java/variety_defaultunsupported/FFoo.java</code> line(s) ['1123', '1339', '1773', '2251', '2666', '3079', '3483', '3942', '4273', '4685', '5041', '5281', '5573', '5803', '6094', '6401', '6707', '7010', '7313', '7616', '7919', '8222', '8525'] added</li>
</details>

## Action Items
* Obtain a [security review](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); reviewer should pay special attention to insights listed above
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: compiler/compiler_test.go [hunk 1]
```diff
@@ -488,6 +488,25 @@ func TestValidJavaWithAsync(t *testing.T) {
...
+	globals.Now = time.Date(2015, 11, 24, 0, 0, 0, 0, time.UTC)
+
+	options := compiler.Options{
+		File:  frugalGenFile,
+		Gen:   "java:default_unsupported",
+		Out:   outputDir + "/defaultunsupported",
+		Delim: delim,
+	}
+	suite := ComparisonList{
+		{"java/variety_defaultunsupported/FFoo.java", "defaultunsupported/variety/java/FFoo.java"},
```
Comment (patkujawa-wf) [inline compiler/compiler_test.go:505]: Wow, that's a big old file! 8k lines :ohmy:

**Review by patkujawa-wf — APPROVED** [review]

**Comment by patkujawa-wf** [comment]
Security +1, no new attack vectors.

**Review by shaunbrockhoff-wk — APPROVED** [review]

**Review by mitchellsinclair-wk — APPROVED** [review]

**Review by danielharasymiw-wf — APPROVED** [review]

**Comment by danielharasymiw-wf** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**